syntax  = "proto3";

package media;


enum MsgType{
    RESPONSE = 0;
    LOGIN = 1;
    PLAY_ONLINE_RANDOM = 2;
    PLAY_ONLINE_RANDOM_RESPONSE = 3;
    DOWNLOAD_SINGLE_MUSIC = 4;
    DOWNLOAD_SINGLE_MUSIC_RESPONSE = 5;
}


message Login {
    string userName = 1;
    string pwd = 2;
}

message Response {
    MsgType cmd = 1;
    int32 code = 2;
    int32 reserved1 = 3;
    string reserved2 = 4;
}

message PlayOnlineRandom {
    string userName = 1;
}

message PlayOnlineRandomRsp{
    repeated string  musicName = 1;
}

message DownloadSingleMusic {
    string  userName = 1;
    string  musicName = 2;
}

message DownloadSingleMusicRsp {
    string musicName = 1;
    int32  fileSize = 2;
    bytes  data = 3;
}